### ISSUE TITLE
Smart Tag option in metadata component does not work.

### ISSUE THREAD

**Issue by tuhinghosh2610** [Adobe-Marketing-Cloud/asset-share-commons#208]
**Describe the bug**
Smart Tag option in metadata component does not work.

* AEM Version: AEM 6.3
* Asset Share Commons Version: 1.5.2
* Author, Publish or both? : Both

**To Reproduce**
Drag and drop metadata component in a asset share commons details page (or any other page).
And then select smart tag option from the dropdown to show smart tags, if they are available smart tag values should be shown.

**Expected behavior**
smart tag values should be shown.

**Comment by davidjgonzalez** [comment]
@tuhinghosh2610 per https://adobe-marketing-cloud.github.io/asset-share-commons/pages/development/computed-properties/ the tagTitles computer property reads from cq:tags (Ie NOT Smart Tags).

But you bring up an interesting topic ... should Smart Tags and AEM Tags display as mixed? I would generally say no since usually, they serve different, though often conflated, purposes.

* AEM Tags are (almost) always a controlled vocabulary, and define taxonomy.
* Smart Tags enrich discovery and are not guaranteed (thus you cannot depend on them for taxonomical control).... even w/ Training Smart Tags.

So, in your opinion, should Smart Tags be selectable as a discrete set of metadata to display? And then, what is the "real" value of them as they simply describe the image.  For example, if there is a Smart Tag of "leaf" .. that is useful in searching for assets that have leaves in them ... but once you get the results, what's the value of the smart tags? I'd argue is ~0 ... if your end-user can't look at a picture and tell its got a leaf in it and has to rely on the Smart Tags text of "leaf" to tell it has leaves in it.. then you have a much bigger problem, no?

The only use case i can think of displaying smart tags is to also wire them up so you can click on and search for things w that keyword (almost like a "more like this" option), but thats something totally different.

So, TL;DR .. whats the business value in displaying smart tags? How will it help users?

**Comment by tuhinghosh2610** [comment]
Hi @davidjgonzalez ,

Thanks for replying back to me. The request to add support for the smart tag came from the client who is keen to display it in the page if present.
The rationale they are giving is, there might be an asset which is tagged with "leaf" by smart tag (not visible) and the user search randomly for the image with "leaf" search term, when the image comes back in result it may not have "leaf" explicitly visible in the image, so the question may come up why the image is relevant for the search.  By presenting the smart tag in the image properties it adds additional transparency for the user regarding the image relevance for the search.
Thus the client is very much keen to display the smart tags. Now since we had to fix this anyhow, Jo suggested why not put it in the product as a PR. I am almost done with the fix, and would be adding a pull request. If you see any value of having the support for smart tag, we could merge it with the ASC code. 

Thanks
Tuhin

**Comment by tuhinghosh2610** [comment]
Hi @davidjgonzalez ,

I have created the PR for you to review (https://github.com/Adobe-Marketing-Cloud/asset-share-commons/pull/209). A shout to Jo (from adobe) for the guidance on this.

**Comment by davidjgonzalez** [comment]
I’ll take it a look - but we’ll need to make sure this doesn’t confuse taxonomy w smart tags.

Tho the rationale is unfortunate and points to smart tags doing a poor job - perhaps the confidence score is low if text is required to explain to a user why a term they selected results in a visual asset.

**Comment by tuhinghosh2610** [comment]
Hi @davidjgonzalez ,

I have updated the code as per your comments, also have refactored it, and looks much more readable now.
**If no properties are selected, then use tagTitles (existing default behavior)** - Working as expected.
**Update dialog to write multiple values to tagPropertyName** - Done
**For each of those defined properties, do a asset.getProperties().get(..) ... (which has the problem we dont actually know the return types... :()** - Doing now, also addressed the problem of the return types by instanceof check.
**for each result from each tagPropertyName, check if the value resolves as a tagId (if it does, assume its a real tag) and get the localized title** - Doing
**If tag == null, then assume its a title, and display that (not sure how to best localize this; i18n?)** - Doing apart from the i18n part, does not make much sense to localise smart tags, should be shown as is if required.
**dump all the titles them all into 1 list that is maybe sortable (alpha a-z?)** - Done, and list is sorted before returning the result.

Kindly let me know about your thoughts. 

Thanks and Regards,
Tuhin

**Comment by davidjgonzalez** [comment]
Released in v1.6.0
